javascript/ql/src/experimental/Security/CWE-918/SSRF.qll

```diff
@@ -0,0 +1,119 @@
+import javascript
+
+import semmle.javascript.security.dataflow.RequestForgeryCustomizations
+import semmle.javascript.security.dataflow.UrlConcatenation
+
+class Configuration extends TaintTracking::Configuration {
+    Configuration() { this = "SSRF" }
+
+    override predicate isSource(DataFlow::Node source) { source instanceof RequestForgery::Source }
+
+    override predicate isSink(DataFlow::Node sink) { sink instanceof RequestForgery::Sink }
+
+    override predicate isSanitizer(DataFlow::Node node) {
+        super.isSanitizer(node) or
+        node instanceof RequestForgery::Sanitizer
+    }
+
+    override predicate isSanitizerEdge(DataFlow::Node source, DataFlow::Node sink) {
+        sanitizingPrefixEdge(source, sink)
+    }
+    override predicate isSanitizerGuard(TaintTracking::SanitizerGuardNode nd) {
+        nd instanceof IntegerCheck or
+        nd instanceof ValidatorCheck or
+        nd instanceof TernaryOperatorSanitizerGuard
+    }
+}
+
+/** TODO add comment */
+class TernaryOperatorSanitizerGuard extends TaintTracking::SanitizerGuardNode {
+    TaintTracking::SanitizerGuardNode originalGuard;
+
+    TernaryOperatorSanitizerGuard() {
+    exists(DataFlow::Node falseNode |
+        this.getAPredecessor+() = falseNode and 
+        falseNode.asExpr().(BooleanLiteral).mayHaveBooleanValue(false)
+    ) and
```

Comment:
You should be able to simplify this to `this.getAPredecessor+().asExpr().(BooleanLiteral).mayHaveBooleanValue(false)`, I think.